core/src/translate/function.rs

```diff
@@ -339,6 +339,14 @@ pub fn translate_function(sql_function: &SqlFunction) -> Result<Expr> {
                 divisor,
             })))
         }
+        "RAND" => {
+            check_len_range(name, args.len(), 0, 1)?;
+            let expr = match args.len() {
+                1 => Some(translate_expr(args[0])?),
+                0 => None,
+            };
```

Comment:
The return type of `len()` is usize.
Using the match syntax will result in unreachable code.
How about writing code like this:

```rust
let expr = args.get(0).map(|expr| translate_expr(expr)).transpose()?;
```